File: simpsons/backend/src/models/Season.ts
import { Schema, model } from 'mongoose';

interface Season {
    season: number;
    title: string;
    year?: number;
}

const SeasonSchema = new Schema<Season>({
    season: { type: Number, required: true },
    title: { type: String, required: true },
    year: { type: Number, required: false }
});

export default model<Season>('Season', SeasonSchema);

File: simpsons/lib/models/season.dart
class Season {
  final String id;
  final int number;
  final String title;

  Season({
    required this.id,
    required this.number,
    required this.title,
  });

  factory Season.fromJson(Map<String, dynamic> json) {
    return Season(
      id: json['_id'] as String,
      number: json['season'] as int,
      title: json['title'] as String,
    );
  }
}
